### PR TITLE
Allow the agent to be removed.

### PR DESCRIPTION
**Pull request by cwood**

### CODE REVIEW

**Comment by ianssoftcom** [comment]
I am also looking for startup to become configurable.  The zabbix-agent is installed by default, but when building ephemeral instances I need to remove zabbix-agent from startup.

File: Gemfile [hunk 1]
```diff
@@ -28,6 +28,7 @@ group :test do
   gem 'voxpupuli-release',                                          :require => false, :git => 'https://github.com/voxpupuli/voxpupuli-release-gem.git'
   gem 'puppet-strings',                                             :require => false, :git => 'https://github.com/puppetlabs/puppetlabs-strings.git'
   gem 'rubocop', '~> 0.39',                                         :require => false
+  gem 'listen', '3.0.5'
```
Comment (bastelfreak) [inline Gemfile]: why do you add the listen gem? it is already present in the development group. (at least sort of, we've guard-rake there which requires listen)

File: manifests/agent.pp [hunk 2]
```diff
@@ -306,6 +307,27 @@
...
+      $file_ensure = 'absent'
+      $directory_ensure = 'absent'
+    }
+    'present': {
+      $service_ensure = 'running'
+      $service_enabled = true
+      $file_ensure = 'file'
+      $directory_ensure = 'directory'
+    }
+    default: {
```
Comment (bastelfreak) [inline manifests/agent.pp]: Removing the 'present' block should be okay here because default is identical?

File: spec/classes/agent_spec.rb [hunk 3]
```diff
@@ -83,6 +83,21 @@
...
+          ensure: "absent",
+          zabbix_package_state: "absent",
+          agent_configfile_path: '/etc/zabbix/zabbix_agentd.conf',
+        }
+      end
+
+      it { should contain_service("zabbix-agent").with_ensure("stopped") }
+      it { should contain_package("zabbix-agent").with_ensure("absent") }
+      it { should contain_file("/etc/zabbix/zabbix_agentd.conf").with_ensure("absent") }
+      it { should contain_file("/etc/zabbix/zabbix_agentd.d").with_ensure("absent") }
```
Comment (bastelfreak) [inline spec/classes/agent_spec.rb]: all these double quotes need to be migrated to single quotes.

File: manifests/agent.pp [hunk 4]
```diff
@@ -334,14 +356,26 @@
...
-    notify  => Service['zabbix-agent'],
-    require => File[$agent_configfile_path],
+  if $directory_ensure == 'directory' {
+    file { $include_dir:
+      ensure  => $directory_ensure,
+      owner   => 'zabbix',
+      group   => 'zabbix',
+      recurse => true,
+      purge   => $include_dir_purge,
+      notify  => Service['zabbix-agent'],
```
Comment (bastelfreak) [inline manifests/agent.pp]: what about creating a $purge variable which is undef or $include_dir_purge and don't duplicate the file resource? You could add that in the upper case block.

**Comment by jyaworski** [comment]
@cwood can you rebase and squash?

**Comment by bastelfreak** [comment]
ping @cwood

**Comment by bastelfreak** [comment]
Thanks for opening this issue. It hasn't seen activity in a while, so I'm going to close the issue. This doesn't mean that we don't think this issue is valid, we're just trying to assess what issues are still relevant. If you still need this(even if you're not willing to work on it yourself) please feel free to reopen it. Again thanks for opening the issue, users who report bugs and ask for features are some of the most important players in the open source community. Feel free to hop by #voxpupuli on irc.freenode.net if you'd like to chat with us about any of this.

**Comment by cwood** [comment]
I only needed it to do what I needed above and remove it all. Its fine if you dont wanna continue this forward.

**Pull request closed without merging**